capi/geos_c.h.in

```diff
@@ -306,19 +306,56 @@ typedef int (*GEOSTransformXYCallback)(
 typedef void (GEOSInterruptCallback)(void);
 
 /**
-* Register a function to be called when processing is interrupted.
+* Callback function for use in interruption. The callback will be invoked at each
+* possible interruption point and can be used to request interruption.
+*
+* \see GEOS_interruptRegisterThreadCallback
+* \see GEOS_interruptThread
+*/
+typedef void (GEOSInterruptThreadCallback)(void*);
```

Comment:
I would avoid the word thread in these new APIs and associated documentation. Maybe this one could be named `GEOSContextInterruptCallback` and similar for other new symbols.